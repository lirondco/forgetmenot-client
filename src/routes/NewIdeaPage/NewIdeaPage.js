import React, { Component } from 'react'
import NewIdeaForm from '../../components/NewIdeaForm/NewIdeaForm'
import { Section } from '../../components/Utils/Utils'

export default class NewIdeaPage extends Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
    }

    handlePostIdeaSuccess = idea => {
        const { history } = this.props
        history.push('/')
    }

    render() {
        return (
            <Section className = 'NewIdeaPage'>
                <NewIdeaForm
                 onPostIdeaSuccess={this.handlePostIdeaSuccess}
                />
            </Section>
        )
    }
}