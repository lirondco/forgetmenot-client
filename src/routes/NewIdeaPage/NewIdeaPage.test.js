import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NewIdeaPage from './NewIdeaPage'

describe(`NewIdeaPage component`, () => {
  it('renders the complete form', () => {
    const wrapper = shallow(<NewIdeaPage />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})