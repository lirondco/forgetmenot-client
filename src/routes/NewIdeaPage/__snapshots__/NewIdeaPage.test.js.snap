// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIdeaPage component renders the complete form 1`] = `
<Section
  className="NewIdeaPage"
>
  <NewIdeaForm
    onPostIdeaSuccess={[Function]}
  />
</Section>
`;
